api/app/signals/apps/reporting/services/clean_up_datawarehouse.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2022 Vereniging van Nederlandse Gemeenten
+import os
+import pathlib
+from datetime import date, timedelta
+
+from django.utils import timezone
+
+
+class BadDataRoot(Exception):
+    pass
+
+
+class DataWarehouseDiskCleaner:
+    """
+    Clean-up disk storage used by datawarehouse CSV dumps.
+
+    Note: this uses can only be used for local disks not for remote file storage
+    like SWIFT storage.
+    """
+    GLOB_PATTERNS = [
+        '[0-9]' * 6 + '*.csv',
+        '[0-9]' * 8 + '_' + '[0-9]' * 6 + '*.zip',
+    ]
+
+    @staticmethod
+    def _data_directories(data_root):
+        return data_root.rglob('[0-9][0-9][0-9][0-9]/[0-9][0-9]/[0-9][0-9]/')
+
+    @staticmethod
+    def _find_data_directories(data_root, start_date, end_date):
+        """
+        Yield data directories, under `data_root` starting at `start_date` and up to and including `end_date`.
+        """
+        if data_root.match('/'):
+            raise BadDataRoot('Cannot look for data exports under filesystem root!')
+        data_directories = DataWarehouseDiskCleaner._data_directories(data_root)
+
+        found = []
+        for path in data_directories:
+            if not path.is_dir():
+                continue
+
+            year_s, month_s, day_s = path.parts[-3:]
+            to_consider = date(int(year_s), int(month_s), int(day_s))
+            if not (start_date <= to_consider <= end_date):
+                continue
+            found.append(path)
+
+        return found
+
+    @staticmethod
+    def _delete_files_from_directory(path, dry_run):
+        """
+        Delete known CSV exports and Zip archives from given data directory.
+        """
```

Comment:
Maybe a good addition is to also be able to select which type of files are being deleted. So csv AND/OR zip files. This way the code is a bit more flexible. For example the CSV files needs to be deleted after one week and the zip files are kept as backup for one month.